test/e2e/gitopsaddon/scripts/setup_env.sh

```
@@ -220,28 +220,47 @@ spec:
+    sleep 5
+  done
 done
+echo "All agent secrets are ready"
+
+# ------- Step 3b2: Patch operator ClusterRole for upstream argocd-operator -------
+# The embedded chart's ClusterRole is extracted from the Red Hat CSV which removed
+# argocdexports. The upstream argocd-operator (used in e2e) still watches ArgoCDExport
+# resources and needs list/watch permission. Without this, the operator's informer cache
+# fails to sync and it never reconciles ArgoCD CRs.
+echo "===== Step 3b2: Patching operator ClusterRole for upstream argocd-operator ====="
+${KUBECTL} --context ${HUB_CTX} patch clusterrole openshift-gitops-operator-manager-role --type=json -p='[
```

Shellcheck (0.11.0)

[info] Double quote to prevent globbing and word splitting.

(SC2086)